### PR TITLE
[DOCS] Adds JSON and JSONParseResult doc

### PR DESCRIPTION
**Pull request by Nutriz**
I saw that GDScript.parse_json and GDScript.to_json is still here, JSON class is not here to replace that

### CODE REVIEW

File: doc/classes/JSON.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="JSON" inherits="Object" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Helper class for parsing JSON.
```
Comment (cbscribe) [inline doc/classes/JSON.xml]: I suggest "for parsing JSON data."

File: doc/classes/JSON.xml [hunk 2]
```diff
@@ -15,6 +17,7 @@
 			<argument index="0" name="json" type="String">
 			</argument>
 			<description>
+				Parse a JSON encoded string and returns a [JSONParseResult] containing the result.
```
Comment (cbscribe) [inline doc/classes/JSON.xml]: Parses

File: doc/classes/JSON.xml [hunk 3]
```diff
@@ -23,6 +26,7 @@
 			<argument index="0" name="value" type="Variant">
 			</argument>
 			<description>
+				Converts a Variant var to JSON text and return the result. Useful for serializing data to store or send over the network.
```
Comment (cbscribe) [inline doc/classes/JSON.xml]: return -> returns

File: doc/classes/JSONParseResult.xml [hunk 4]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="JSONParseResult" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Data class wrapper for decoded JSON.
 	</brief_description>
 	<description>
+		Returned by [method JSON.parse], [JSONParseResult] contains decoded JSON or error information if JSON source not successfull parsed.	Check if JSON source successfull parsed with [code]if json_result.error == 0[/code].
```
Comment (cbscribe) [inline doc/classes/JSONParseResult.xml]: "if JSON source was not successfully parsed."
Too many spaces after second sentence.
Last sentence:
"You can check if JSON source was successfully parsed"

File: doc/classes/JSONParseResult.xml [hunk 5]
```diff
@@ -68,12 +70,24 @@
 	</methods>
 	<members>
 		<member name="error" type="int" setter="set_error" getter="get_error" enum="Error">
+			Returns error's type if JSON source not successfull parsed. See [@Global Scope].Error.
```
Comment (cbscribe) [inline doc/classes/JSONParseResult.xml]: In general, member descriptions don't say "Returns". This is a value, not a function, and it usually can both set and get.
Try:
"The error type if if JSON source was not successfully parsed." 

That link at the end isn't right.  I don't think we have constant links, so you have to say something like "See [@Global Scope] ERR_* constants."

File: doc/classes/JSONParseResult.xml [hunk 6]
```diff
@@ -68,12 +70,24 @@
...
 		</member>
 		<member name="error_line" type="int" setter="set_error_line" getter="get_error_line">
+			Returns error's line if JSON source not successful parsed.
```
Comment (cbscribe) [inline doc/classes/JSONParseResult.xml]: The line number where the error occurred if JSON source was not successfully parsed.

File: doc/classes/JSONParseResult.xml [hunk 7]
```diff
@@ -68,12 +70,24 @@
...
 		</member>
 		<member name="error_string" type="String" setter="set_error_string" getter="get_error_string">
+			Returns error message if JSON source not successful parsed. See [@Global Scope].Error.
```
Comment (cbscribe) [inline doc/classes/JSONParseResult.xml]: The error message if JSON source was not successfully parsed.
See above re: link.

File: doc/classes/JSONParseResult.xml [hunk 8]
```diff
@@ -68,12 +70,24 @@
...
 		</member>
 		<member name="result" type="Variant" setter="set_result" getter="get_result">
+			Returns a [Variant] containing the parsed JSON. Use typeof() to check if it is what you expect. For exemple, if JSON source starts with braces [code]{}[/code] a [Dictionary] will be returned, if JSON source starts with array braces [code][][/code] an [Array] will be returned.
```
Comment (cbscribe) [inline doc/classes/JSONParseResult.xml]: A [Variant] containing...

**Review by cbscribe — CHANGES_REQUESTED** [review]

**Comment by Jerome67000** [comment]
@cbscribe updated 👍

File: doc/classes/JSONParseResult.xml [hunk 9]
```diff
@@ -68,12 +70,24 @@
 	</methods>
 	<members>
 		<member name="error" type="int" setter="set_error" getter="get_error" enum="Error">
+			The error type if JSON source was not successfully parsed. See [@Global Scope]ERR_* constants.
 		</member>
 		<member name="error_line" type="int" setter="set_error_line" getter="get_error_line">
+			The line number where the error occurred if JSON source was not successfully parsed.
 		</member>
 		<member name="error_string" type="String" setter="set_error_string" getter="get_error_string">
+			The line number where the error occurred if JSON source was not successfully parsed. See [@Global Scope]ERR_* constants.
```
Comment (cbscribe) [inline doc/classes/JSONParseResult.xml]: Shouldn't this be the error string?

**Comment by Jerome67000** [comment]
@cbscribe fixed :)

File: doc/classes/JSONParseResult.xml [hunk 10]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="JSONParseResult" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Data class wrapper for decoded JSON.
 	</brief_description>
 	<description>
+		Returned by [method JSON.parse], [JSONParseResult] contains decoded JSON or error information if JSON source not successfull parsed. You can check if JSON source was successfully parsed with [code]if json_result.error == 0[/code].
```
Comment (Noshyaar) [inline doc/classes/JSONParseResult.xml]: successfull typo

**Comment by Jerome67000** [comment]
@Noshyaar Done ! Just intalled a spellcheck for Atom. Hop it will help with this type of errors.

**Review by ghost — APPROVED** [review]

**Review by cbscribe — APPROVED** [review]

**Pull request merged**